File: php/src/Model/ArrayProperty.php
<?php

declare(strict_types=1);

namespace Oojs\Model;

/**
 * A property whose value is a JSON array with homogeneous items.
 * Nested arrays are not supported in v1.0.
 */
class ArrayProperty implements PropertyDef
{
    public function __construct(
        public readonly PropertyDef $items,
        public readonly int $minItems = 0,
        public readonly ?int $maxItems = null,
        public readonly bool $uniqueItems = false,
        public readonly string $title = '',
        public readonly string $description = '',
    ) {}
}

File: java/src/main/java/org/oojs/model/ArrayProperty.java
package org.oojs.model;

/** A property whose value is a JSON array with homogeneous items. Nested arrays are not supported in v1.0. */
public final class ArrayProperty implements PropertyDef {
    public final PropertyDef items;
    public final int minItems;
    public final Integer maxItems;
    public final boolean uniqueItems;
    public final String title;
    public final String description;

    public ArrayProperty(PropertyDef items, int minItems, Integer maxItems,
            boolean uniqueItems, String title, String description) {
        this.items = items;
        this.minItems = minItems;
        this.maxItems = maxItems;
        this.uniqueItems = uniqueItems;
        this.title = title;
        this.description = description;
    }
}
